doc/release-notes/release-7.sgml.in

```diff
@@ -97,7 +97,9 @@ This section gives an account of those changes in three categories:
 <sect1>Removed options<label id="removedoptions">
 <p>
 <descrip>
-	<p>No removed options in this version.
+	<tag>CPPFLAGS=-DHEADERS_LOG</tag>
+	<p>The code enabled by this preprocessor macro has not built for many
+	   years, indicating that the feature is unused.
```

Comment:
> Please add text to `doc/release-notes/release-7.sgml.in` for the remove `./configure options` section saying:
> 
> ```
> <tag>-DHEADERS_LOG</tag>
> <p>Removed. This code has not built for many years, indicating it is not being used by anyone.
> ```

I added similar text in branch commit 07c2b03, modeling after `-DUSE_CHUNKEDMEMPOOLS=1` in doc/release-notes/release-5.sgml.in and correcting for the fact that `-D` is not really a `./configure` option. Please adjust further if you insist.

FWIW, I do not think we should document removal of secret hacks like `HEADERS_LOG` in release notes, especially hacks that were dead for many years. This information is just noise for any release notes reader that should not be reading `git log`.

I also adjusted PR title after your PR title changes. I disagree with your removal of the `Maintenance` prefix, but I did not add it back in.